apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/helpers/Builder.kt

```diff
@@ -0,0 +1,240 @@
+package com.apollographql.apollo3.compiler.codegen.java.helpers
+
+import com.apollographql.apollo3.compiler.codegen.ClassNames
+import com.apollographql.apollo3.compiler.codegen.java.JavaAnnotations
+import com.apollographql.apollo3.compiler.codegen.java.JavaClassNames
+import com.apollographql.apollo3.compiler.codegen.java.JavaContext
+import com.apollographql.apollo3.compiler.decapitalizeFirstLetter
+import com.apollographql.apollo3.compiler.defaultOptionalValue
+import com.apollographql.apollo3.compiler.ir.IrEnumValue
+import com.apollographql.apollo3.compiler.ir.IrListValue
+import com.apollographql.apollo3.compiler.ir.IrValue
+import com.apollographql.apollo3.compiler.isList
+import com.apollographql.apollo3.compiler.isOptional
+import com.apollographql.apollo3.compiler.listParamType
+import com.apollographql.apollo3.compiler.unwrapOptionalType
+import com.apollographql.apollo3.compiler.wrapOptionalValue
+import com.squareup.javapoet.ClassName
+import com.squareup.javapoet.CodeBlock
+import com.squareup.javapoet.FieldSpec
+import com.squareup.javapoet.MethodSpec
+import com.squareup.javapoet.ParameterSpec
+import com.squareup.javapoet.ParameterizedTypeName
+import com.squareup.javapoet.TypeName
+import com.squareup.javapoet.TypeSpec
+import javax.lang.model.element.Modifier
+
+internal class Builder(
+  val targetObjectClassName: ClassName,
+  val fields: List<Pair<String, TypeName>>,
+  val fieldDefaultValues: Map<String, IrValue?>,
+  val fieldJavaDocs: Map<String, String>,
+  val buildableTypes: List<TypeName> = emptyList(),
+  val context: JavaContext
+) {
+  fun build(): TypeSpec {
+    return TypeSpec.classBuilder(JavaClassNames.Builder.simpleName())
+      .addModifiers(Modifier.PUBLIC, Modifier.STATIC, Modifier.FINAL)
+      .addFields(builderFields())
+      .addMethod(MethodSpec.constructorBuilder().build())
+      .addMethods(fieldSetterMethodSpecs())
+      .addMethods(inputFieldSetterMethodSpecs())
+      .addMethods(fieldSetterWithMutatorMethodSpecs())
+      .addMethod(buildMethod())
+      .build()
+  }
+
+  private fun builderFields(): List<FieldSpec> {
+    fun valueCode(value: IrValue, type: TypeName): CodeBlock {
+      return when (value) {
+        is IrEnumValue -> CodeBlock.of("\$T.safeValueOf(\$S)", type, value.value)
+        else -> value.codeBlock()
```

Comment:
The new `codeBlock()` method allowed the original code to be greatly simplified. Only problem is `IrEnumValue.codeBlock()` doesn't work because it doesn't have access to the type name.

Reply:
I think this all goes aways because we don't want default values in Kotlin code?

Reply:
True. Eventually. 😃 My thought was to port over everything intact, and then later rip out all default value code everywhere in the whole project.

Reply:
Sounds like a plan!

Reply:
I went ahead and removed all the code that assigns a default value to builder fields so we can have a clean PR to merge into main.